src/tools/generate_insights.py

```diff
@@ -39,139 +63,206 @@ class Insight(BaseModel):
+        data_csv = raw_data.to_csv(index=False)
+        logger.debug(f"Data columns: {list(raw_data.columns)}")
+    else:
+        data_csv = str(raw_data)
+
+    # Generate insights using the LLM
+    try:
+        chain = INSIGHT_GENERATION_PROMPT | sonnet.with_structured_output(InsightResponse)
+        response = chain.invoke({
             "user_query": query,
-            "raw_data": df.to_csv(index=False),
-        }
-    )
-    logger.debug(f"Successfully generated insight: '{insight.title}'")
+            "raw_data": data_csv
+        })
 
-    # Convert dataframe to markdown for the insight object
+        insights = response.insights
+        logger.debug(f"Generated {len(insights)} insights")
 
-    tool_message = ToolMessage(
-        content=f"Successfully generated an insight: '{insight.title}'. Raw data: {df.to_csv(index=False)}",
-        tool_call_id=tool_call_id,
-    )
+        # Format the response message
+        message_parts = []
+        charts_data = []
+        
+        for i, insight in enumerate(insights, 1):
+            logger.debug(f"Insight {i}: {insight.title} ({insight.chart_type})")
+            
+            message_parts.append(f"**Insight {i}: {insight.title}**")
+            message_parts.append(f"Chart Type: {insight.chart_type}")
+            message_parts.append(f"Key Finding: {insight.insight}")
+            message_parts.append(f"Data Points: {len(insight.data)}")
+            message_parts.append("")
+            
+            # Store chart data for frontend
+            charts_data.append({
+                "id": f"chart_{i}",
+                "title": insight.title,
+                "type": insight.chart_type,
+                "insight": insight.insight,
+                "data": insight.data,
+                "xAxis": insight.x_axis,
+                "yAxis": insight.y_axis,
+                "colorField": insight.color_field
+            })
 
-    return Command(
-        update={
-            "insights": insight.model_dump(),
-            "messages": [tool_message],
-        },
-    )
+        tool_message = "\n".join(message_parts)
+        
+        # Update state with generated insights
+        updated_state = {
+            "insights": response.model_dump()["insights"],
+            "charts_data": charts_data,
+            "insight_count": len(insights)
+        }
+
+        return Command(
+            update={
+                **updated_state,
+                "messages": [
+                    ToolMessage(
+                        content=tool_message,
+                        tool_call_id=tool_call_id,
+                    )
+                ],
+            }
+        )
+
+    except Exception as e:
+        error_msg = f"Error generating insights: {str(e)}"
+        logger.error(error_msg, exc_info=True)
+        return Command(
+            update={
+                "messages": [
+                    ToolMessage(
+                        content=error_msg,
```

Comment:
Same here, if this is the generate-insights response it has to be caught separately by the frontend.